class.jetpack-jitm.php

```diff
@@ -43,27 +44,42 @@ private function __construct() {
-		if ( 'media-new.php' == $pagenow && ! Jetpack::is_module_active( 'photon' ) && 'hide' != $showphoton ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-			add_action( 'post-plupload-upload-ui', array( $this, 'photon_msg' ) );
+		// so if it's not an array, it means no JITM was dismissed
+		if ( ! is_array( self::$jetpack_hide_jitm ) ) {
+			if ( 'media-new.php' == $pagenow && ! Jetpack::is_module_active( 'photon' ) ) {
+				add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
+				add_action( 'post-plupload-upload-ui', array( $this, 'photon_msg' ) );
+			}
+			else if ( 'update-core.php' == $pagenow && ! $auto_updates_disabled && ! Jetpack::is_module_active( 'manage' ) ) {
+				add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
+				add_action( 'admin_notices', array( $this, 'manage_msg' ) );
+			}
+			elseif ( 'plugins.php' == $pagenow && ! $auto_updates_disabled && ( isset( $_GET['activate'] ) && 'true' === $_GET['activate'] || isset( $_GET['activate-multi'] ) && 'true' === $_GET['activate-multi'] ) ) {
+				add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
+				add_action( 'pre_current_active_plugins', array( $this, 'manage_pi_msg' ) );
+			}
+			elseif ( 'post-new.php' == $pagenow && in_array( $screen->post_type, array( 'post', 'page' ) ) ) {
+				add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
+				add_action( 'admin_notices', array( $this, 'editor_msg' ) );
+			}
 		}
-		else if ( 'update-core.php' == $pagenow && 'hide' != $showmanage ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-			add_action( 'admin_notices', array( $this, 'manage_msg' ) );
-		}
-		elseif ( 'plugins.php' === $pagenow && isset( $_GET['activate'] ) && 'true' === $_GET['activate'] && 'hide' != $show_manage_pi ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-			add_action( 'pre_current_active_plugins', array( $this, 'manage_pi_msg' ) );
-		}
-		elseif ( 'post-new.php' === $pagenow && in_array( $post_type, array( 'post', 'page' ) ) && 'hide' != $show_editor ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-			add_action( 'admin_notices', array( $this, 'editor_msg' ) );
+	}
+
+	/**
+	 * Save plugins that are activated. This is used when one or more plugins are activated to know
+	 * what was activated and use it in Jetpack_JITM::manage_pi_msg() before deleting the option.
+	 *
+	 * @since 3.8.2
+	 */
+	function previously_activated_plugins() {
+		$wp_list_table = _get_list_table( 'WP_Plugins_List_Table' );
+		$action = $wp_list_table->current_action();
+		if ( $action && ( 'activate' == $action || 'activate-selected' == $action ) ) {
+			update_option( 'jetpack_previously_activated', get_option( 'active_plugins', array() ) );
```

Comment:
Because we need to know which plugins were activated. We get `active_plugins` options before activating the plugins, and again after the plugins are activated. We then get the difference, which will be the plugins that were activated.